contract/services.py

```diff
@@ -351,6 +354,50 @@ def __copy_details(self, contract_id, amended_contract):
             cd_new.json_ext = None
             cd_new.save(username=self.user.username)
 
+    @check_authentication
+    def renew(self, contract):
+        try:
+            # check rights for delete contract
+            if not self.user.has_perms(ContractConfig.gql_mutation_renew_contract_perms):
+                raise PermissionError("Unauthorized")
+            from core import datetime, datetimedelta
+            contract_to_renew = ContractModel.objects.filter(id=contract["id"]).first()
+            contract_id = contract["id"]
+            # block deleting contract not in Updateable or Approvable state
+            state_right = self.__check_rights_by_status(contract_to_renew.state)
+            # check if we can amend
+            if state_right is not "cannot_update" and contract_to_renew.state is not ContractModel.STATE_TERMINATED:
+                raise ContractUpdateError("You cannot renew this contract!")
+            # create copy of the contract
+            renewed_contract = copy(contract_to_renew)
```

Comment:
if PH is set, we might get the PH insuree again  (this is a should, at least add a todo)

Reply:
if other cases will the contract details be copied ?

Reply:
Yes, I also copied the contract details so as to have PHInsuree (insuree). Below I also copied the ContractDetails if they are in previous contract. I had to do as a separate operation because as a default you cannot copy related object and below I call a function to copy contract details.

Reply:
@delcroip line 391 - __copy_details function to copy contract details from previous contract

Reply:
@delcroip And also I fixed the comments and made some changes.